pkg/plugin/validation.go

```diff
@@ -17,32 +17,64 @@ limitations under the License.
 package plugin
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/blang/semver"
 
 	"sigs.k8s.io/kubebuilder/pkg/internal/validation"
 )
 
+// errInvalidVersion should be returned when a plugin's version is invalid.
+type errInvalidVersion struct {
+	version, msg string
+}
+
+func (e errInvalidVersion) Error() string {
+	if e.version == "" {
+		return fmt.Sprintf("plugin version is empty")
+	}
+	return fmt.Sprintf("invalid plugin version %q: %s", e.version, e.msg)
+}
+
 // ValidateVersion ensures version adheres to the plugin version format,
 // which is tolerant semver.
 func ValidateVersion(version string) error {
 	if version == "" {
-		return errors.New("plugin version is empty")
+		return errInvalidVersion{}
 	}
 	// ParseTolerant allows versions with a "v" prefix or shortened versions,
 	// ex. "3" or "v3.0".
-	if _, err := semver.ParseTolerant(version); err != nil {
-		return fmt.Errorf("failed to validate plugin version %q: %v", version, err)
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		return errInvalidVersion{version, err.Error()}
+	}
+	cv := semver.Version{Major: v.Major, Minor: v.Minor}
+	if !v.Equals(cv) {
+		return errInvalidVersion{version, "must contain major and minor version only"}
```

Comment:
The plugin version in the master is `v2.0.0` so should it not be failing in the CI to run the script which generates the testdata?

Reply:
Good catch, this should be done using string comp.